libs/ml/include/ml/graph.hpp

```diff
@@ -77,6 +84,18 @@ class Graph : public ops::Trainable<T>
     trainable_[name] = op;
   }
 
+  template <class OperationType>
+  meta::IfIsGraph<ArrayType, OperationType, void> AddTrainable(
```

Comment:
lets move the implementations out of the class body. Also one of your AddTrainable functions is missing comments